R/make_split_fun.R

```diff
@@ -356,15 +367,43 @@ make_split_fun <- function(pre = list(), core_split = NULL, post = list()) {
 #' @export
 #' @family make_custom_split
 add_combo_facet <- function(name, label = name, levels, extra = list()) {
-  function(ret, spl, .spl_context, fulldf) {
-    val <- LevelComboSplitValue(val = name, extr = extra, combolevels = levels, label = label)
-    add_to_split_result(ret,
-      values = list(val), labels = label,
-      datasplit = list(do.call(rbind, ret$datasplit[levels]))
-    )
+    function(ret, spl, .spl_context, fulldf) {
+        if(is(levels, "AllLevelsSentinel")) {
+            subexpr <- expression(TRUE)
+            datpart <- list(fulldf)
+        } else {
+            subexpr <- Reduce(.or_combine_exprs, lapply(ret$value[levels], value_expr))
+            datpart <- list(do.call(rbind, ret$datasplit[levels]))
+        }
+
+
+        val <- LevelComboSplitValue(val = name, extr = extra, combolevels = levels, label = label,
+                                    sub_expr = subexpr)
+        add_to_split_result(ret,
+           values = list(val), labels = label,
+           datasplit = datpart
+        )
   }
 }
 
+
+.or_combine_exprs <- function(ex1, ex2) {
```

Comment:
I remember us talking about making expression handling more general and contained in a single system of functions. Do you think it is worth it?

Reply:
I mean adding this to make_subset_expr (or is this too specific?)